assets/js/amp-post-meta-box.js

```diff
@@ -0,0 +1,57 @@
+/* exported AmpPostMetaBox */
+
+/**
+ * AMP Post Meta Box.
+ *
+ * @since 0.6
+ */
+var AmpPostMetaBox = ( function( $ ) {
+	'use strict';
+
+	// Exports.
+	return {
+		/**
+		 * Holds data.
+		 *
+		 * @since 0.6
+		 */
+		data: {},
+
+		/**
+		 * Boot plugin.
+		 *
+		 * @since 0.6
+		 * @param {Object} data Object data.
+		 * @return {void}
+		 */
+		boot: function( data ) {
+			this.data = data;
+
+			$( document ).ready( function() {
+				this.addPreviewButton();
+			}.bind( this ) );
+		},
+
+		/**
+		 * Add AMP Preview button.
+		 *
+		 * @since 0.6
+		 * @return {void}
+		 */
+		addPreviewButton: function() {
+			var $previewBtn = $( '#preview-action a.preview' );
+
+			$previewBtn
+				.clone()
+				.insertAfter( $previewBtn )
+				.addClass( 'amp-preview' )
+				.prop( {
+					'href': this.data.previewLink,
+					'id': 'amp-' + $previewBtn.prop( 'id' ),
+					'target': 'amp-' + $previewBtn.prop( 'target' )
+				} )
```

Comment:
This isn't working yet with autosave revisions. If you try clicking the AMP preview before an autosave happens, then the opened window will not show the changes you've made.

This logic needs to be incorporated: https://github.com/WordPress/wordpress-develop/blob/a825a181e11b1a89e76d79d22bd2c220f45b3741/src/wp-admin/js/post.js#L388-L418

Reply:
Good catch! To prevent rewriting core functionalities and apply DRY principals, I added some logic to trigger the Core Preview but but append the amp query var at the end if the AMP preview button is clicked. That way the events propagate correctly and it also keeps the preview in the same target which is probably better from a user experience perspective.

Reply:
@westonruter [this commit](https://github.com/Automattic/amp-wp/pull/812/commits/779f76317301c3ae1843936d697b58b92cc2bd16) address this issue, it should be working as expected now.